src/ssh/cockpitsshrelay.c

```diff
@@ -641,7 +604,7 @@ set_knownhosts_file (CockpitSshData *data,
    * a security check (if one can write /.ssh/known_hosts then we have to trust them), just caution */
   if (!host_known && g_strcmp0 (g_get_home_dir (), "/") != 0)
     {
-      host_known = cockpit_is_host_known (data->user_known_hosts, host, port);
+      host_known = session_has_known_host_in_file (data->user_known_hosts, data, host, port);
       if (host_known)
```

Comment:
I really think/hope we can simplify `set_knownhosts_file()` a bit, but this can/should be a separate PR.